src/app/creation/creation.component.ts

```diff
@@ -65,9 +97,9 @@ export class CreationComponent implements OnInit {
   }
 
   public saveCharacter() {
-    this.character.name = this.characterName;
-    this.character.race = this.selectedRace.name;
-    this.character.class = this.selectedClass.name;
+    this.character.name = this.characterName ? this.characterName : '';
```

Comment:
nit: you can use a logical OR here:
`this.character.name = this.characterName || '';`